lib/WebServer/MiLightHttpServer.cpp

```diff
@@ -11,74 +11,31 @@
 void MiLightHttpServer::begin() {
   applySettings(settings);
 
-  server.on("/", HTTP_GET, handleServe_P(index_html_gz, index_html_gz_len));
-  server.on("/settings", HTTP_GET, [this]() { serveSettings(); });
-  server.on("/settings", HTTP_PUT, [this]() { handleUpdateSettings(); });
-  server.on("/settings", HTTP_POST,
-    [this]() {
-      Settings::load(settings);
-      server.send_P(200, TEXT_PLAIN, PSTR("success."));
-    },
-    handleUpdateFile(SETTINGS_FILE)
-  );
-  server.on("/radio_configs", HTTP_GET, [this]() { handleGetRadioConfigs(); });
+  // set up HTTP end points to serve
+
+  _handleRootPage = handleServe_P(index_html_gz, index_html_gz_len);
+  server.on("/", HTTP_GET, [this]() { if (server.validateAuthentiation()) _handleRootPage(); });
```

Comment:
Can we pull the auth handling wrapper into a method so that the conditional isn't repeated?  I think probably the easiest way is to just add private methods for `on` and `onPattern` and use those in place of `server.on`, etc.

Example here, in case that's helpful:

https://github.com/sidoh/epaper_templates/blob/master/lib/HTTP/EpaperWebServer.cpp#L408-L429

This is using a different HTTP server, but should be the same idea.